src/mito2/src/flush.rs

```diff
@@ -451,6 +453,17 @@ impl FlushScheduler {
         flush_status.on_failure(Arc::new(RegionClosedSnafu { region_id }.build()));
     }
 
+    /// Notifies the scheduler that the region is truncating.
+    pub(crate) fn on_region_truncating(&mut self, region_id: RegionId) {
+        // Remove this region.
+        let Some(flush_status) = self.region_status.remove(&region_id) else {
+            return;
+        };
+
+        // Notifies all pending tasks.
+        flush_status.on_failure(Arc::new(RegionTruncatingSnafu { region_id }.build()));
+    }
```

Comment:
suggestion: `on_region_dropped/closed/truncating` are similar. Could we add a helper method for them?
```rust
pub(crate) fn on_region_truncating(&mut self, region_id: RegionId) {
    self.remove_region_on_failure(RegionTruncatingSnafu { region_id }.build());
}

fn remove_region_on_failure(&mut self, region_id: RegionId, err: Error) {}
```